### DISCUSSION TITLE
useContractReads not returning newest data

### DISCUSSION THREAD

**Discussion by ArturStaron** [wevm/wagmi#2154] [Q&A]
Hi, I've been trying to use `useContractReads` to get live information from blockchain to have correct project and pricing information while people are minting. However, it seems like whenever I update price on my minter contract, through etherscan, I need to refresh the page to get any updates. I'm working on goerli testnet now but I assume that shouldn't impact how the hook behaves... 

My implementation boils down to:
```
const { data, isError, isLoading } = useContractReads({
    scopeKey: `mintProgressData-${projectId}`,
    contracts: [
      {
        address: tenderContract,
        abi: GenArt721CoreABI,
        functionName: "projectStateData",
        args: [BigNumber.from(projectId)],
      },
      {
        address: tenderMinterContract,
        abi: GenArt721MintABI,
        functionName: "getPriceInfo",
        args: [BigNumber.from(projectId)],
      },
      {
        address: tenderMinterContract,
        abi: GenArt721MintABI,
        functionName: "projectConfig",
        args: [BigNumber.from(projectId)],
      },
    ],
    watch: true,
    cacheTime: 2_000,
    structuralSharing: (prev, next) => {
      return deepEqual(prev, next) ? prev || next : next;
    },
    onSuccess(data) {
      setProjectData(data[0]);
      setProjectPrice(data[1]);
      setProjectAuction(data[2]);
    },
  });
  ```
Which is basically a copy of https://github.com/ArtBlocks/artblocks-engine-react/blob/main/src/components/MintingInterface.tsx
Initially I didn't have `cacheTime` and `structuralSharing` defined, leaving it as default, and hook wasn't working either way. B logging prev and next in console I could see that `structuralSharing` fires every now and then but the neither prev nor next is up to date or changed in any way. Even when waiting significant amount of time after price per token update in minter contract, which should be reflected in `getPriceInfo`.
  
  My app is wrapped in `<WagmiConfig client={wagmiClient}>`
  `wagmiClient` prep code looks like this:
  ```
  const expectedChains = !isEthTestnetEnabled
    ? [chain.mainnet]
    : [chain.goerli];
  const [queryClient] = useState(() => new QueryClient());
  const { provider: wagmiProvider, webSocketProvider: wagmiWebSocketProvider } =
    configureChains(expectedChains, [
      infuraProvider({
        apiKey: INFURA_KEY,
        priority: 0,
      }),
      publicProvider({ priority: 1 }),
    ]);

  const wagmiClient = createClient({
    autoConnect: true,
    provider: wagmiProvider,
    webSocketProvider: wagmiWebSocketProvider,
  });
  ```
  I'm using Infura provider and I can see in Infura dashboard that connection is established whenever the page loads. 
  Am I missing something obvious? I thought even setting `watch` to true would be enough to get new data whenever it updates on chain...
  Any help would be greatly appreciated :)

**Comment by tmm** [comment]
You could set the [`watch: true`](https://wagmi.sh/react/hooks/useContractReads#watch-optional) or manually call `refetch` to fetch the latest data.

**Reply by ArturStaron** [reply]
The issue isn't the lack of `watch: true`. It's set, as visible in the code above. The problem is that we're not receiving the newest data, neither when using websockets nor when removing websocket provider (so that it uses pooling interval). Manually calling `refetch` does work but that doesn't explain why default `watch` behavior doesn't. Ideally we'd have working websockets.
I suppose websocket connection might be broken since we're developing on nextjs. At least based on the seem to be some compatibility problems with ssr: https://github.com/wagmi-dev/wagmi/issues/542 . Are there any plans to get nextjs support?
